daemon/emer-persistent-cache.c

```diff
@@ -459,7 +459,13 @@ update_boot_offset (EmerPersistentCache *self,
         return save_timing_metadata (self, &relative_time, &absolute_time, NULL,
                                      NULL, NULL, error);
 
-      return TRUE;
+      /*
+       * Always make sure a file exists with the boot offset to maintain the
+       * invariant that when this function returns TRUE, a keyfile exists on
+       * disk.
+       */
+      return save_timing_metadata (self, NULL, NULL, &priv->boot_offset,
```

Comment:
The reason the test fails is written in the commit message, isn't it?
On one hand the persistent cache code does not monitor the keyfile, and when the test unlinks it, this method will just return TRUE if the offset is already initialized.
On the other hand the test expects the file to be re-created every time `get_boot_offset()` is created. If we don't do this, we should remove that expectation from the test.